### PR TITLE
#44 progress bar for cells and batch

### PR DESCRIPTION
**Pull request by Adamage**
Cells and batches now report their progress live.

### CODE REVIEW

File: sst/src/exporter/execute_preprocessor_with_progress_bar.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from tqdm.contrib import tenumerate
+
+
+class ExecutePreprocessorWithProgressBar(ExecutePreprocessor):
+    def __init__(self, **kw):
+        super().__init__(**kw)
+
+    def preprocess(self, nb, resources=None, km=None):
+        nb.cells = list(map(itemgetter(1), tenumerate(nb.cells, desc=f"Cells: ", position=0, leave=False)))
+        return super().preprocess(nb=nb, resources=resources, km=km)
```
Comment (plutasnyy) [inline sst/src/exporter/execute_preprocessor_with_progress_bar.py:14]: How clever!

Comment (plutasnyy) [reply sst/src/exporter/execute_preprocessor_with_progress_bar.py:14]: Although I wonder if it makes sense to use tenumerate and take the second element,
what about just 
`nb.cells = list(tqdm(nb.cells, desc=f"Cells: ", position=0, leave=False))`
it won't work?

**Review by plutasnyy — APPROVED** [review]

File: sst/src/exporter/execute_preprocessor_with_progress_bar.py (hunk 1 above)
Comment (plutasnyy) [reply sst/src/exporter/execute_preprocessor_with_progress_bar.py:14]: Or even shorter `nb.cells = tqdm(nb.cells, desc=f"Cells: ", position=0, leave=False)` ?

Comment (Adamage) [reply sst/src/exporter/execute_preprocessor_with_progress_bar.py:14]: So the thing is, `tqdm` is a wrapper that at the same time has a iterable-like behavior. However the `nb.cells` collection needsd to support item addition etc, and be something like 100% Liskov substitutable. So `tqdm` is not.

Basically here I can use one of the three wrappers they provide. I can try and use the other one `tmap` and see if I can make it simpler

**Review by plutasnyy — APPROVED** [review]

File: sst/src/exporter/execute_preprocessor_with_progress_bar.py (hunk 1 above)
Comment (Adamage) [reply sst/src/exporter/execute_preprocessor_with_progress_bar.py]: In the end I extended `list` to handle `tqdm` wrapper separately but only when iterating. In other aspects this new class behaves like a list - although it has a lot of functionality missing. It is not a drag-and-drop replacement for `list`

**Pull request merged**